frontend/src/views/WheresMyGene/components/HeatMap/utils.ts

```diff
@@ -224,6 +224,35 @@ interface CreateYAxisOptionsProps {
   cellTypeIdsToDelete: string[];
 }
 
+/**
+ * Used to calculate text pixel widths. Should be only created once.
+ */
+const CTX = document.createElement("canvas").getContext("2d");
+
+/**
+ * Truncates the string to a given pixel width.
+ *
+ * @param text The text to truncate
+ * @param maxLength The max width in pixels the string should be
+ * @param font The font family and font size as a string. Ex. "12px sans-serif"
+ * @returns The string fixed to a certain pixel width
+ */
+export function getFixedWidth(
+  text: string,
+  maxLength: number,
+  font = "12px sans-serif"
+): string {
+  CTX!.font = font;
+  for (let i = 0; i < text.length; i++) {
+    const substring = text.substring(0, i);
+    const textMetrics = CTX!.measureText(substring);
+    if (textMetrics!.width >= maxLength) {
+      return substring + "...";
```

Comment:
@signechambers1 If we're creating a custom truncation solution for cell type names. Should we follow the explorer and use middle truncation?

Reply:
yes! that would be great for consistency